practice/completer/site.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+/**
+ * Completer object contains completions.
+ *
+ * completions will have keys as the completion and values as weights.
+ */
+function Completer() {
+  this.completions = {};
+}
+
+/**
+ * modifyCompleter has these methods:
+ *
+ * addCompletion() adds a completion to the completions container (in-place)
+ * removeCompletion() removes a completion from the container (in-place)
+ * complete() returns an array of completions, highest weighted first
+ * selectCompletion() increases the weight of a given completion (in-place)
+ *
+ */
+var modifyCompleter = {
+  addCompletion: function(str) {
+    this.completions[str] = 1;
+  },
+  removeCompletion: function(str) {
+    delete this.completions[str];
+  },
+  complete: function(prefix) {
+    var re = RegExp('^' + prefix);
+    var matches = _.pickBy(this.completions,
+      function(v, k) {return re.test(k);});
+    var sortedMatches = _.map(_.orderBy(_.toPairs(matches),
```

Comment:
This variable creation is really compact, but also difficult to unpack what transformation is going on. It might be clearer if it's broken up into a few steps, like
```js
var matchesToPairs = _.toPairs(matches)
var orderedMatches = _.orderBy(matchesToPairs, function(o) {return o[1];}. 'desc'),
var sortedMatches = _.map(orderedMatches(orderedMatches, function(o) {return o[0]});
```
This may be getting a bit verbose, and how to break it down and the variable names may be best left to your judgment, but I think it makes the order of transformation clearer overall.
```